ocean_lib/assets/test/test_utils.py

```diff
@@ -41,13 +45,41 @@ def test_add_trusted_algorithm(publisher_ocean_instance):
         [algorithm_ddo.did], publisher_ocean_instance.config.aquarius_url
     )
 
+    # add a new trusted algorithm to the publisher_trusted_algorithms list
     new_publisher_trusted_algorithms = add_publisher_trusted_algorithm(
         ddo.did, algorithm_ddo_v2.did, publisher_ocean_instance.config.aquarius_url
     )
 
     assert new_publisher_trusted_algorithms is not None
     assert len(new_publisher_trusted_algorithms) > len(publisher_trusted_algorithms)
 
+    # add an existing algorithm to publisher_trusted_algorithms list
+    new_publisher_trusted_algorithms = add_publisher_trusted_algorithm(
+        ddo.did, algorithm_ddo.did, publisher_ocean_instance.config.aquarius_url
+    )
+    assert new_publisher_trusted_algorithms is not None
+    for _, trusted_algorithm in enumerate(publisher_trusted_algorithms):
+        assert trusted_algorithm["did"] == algorithm_ddo.did
+    assert len(new_publisher_trusted_algorithms) == len(publisher_trusted_algorithms)
+
+    # remove an existing algorithm to publisher_trusted_algorithms list
+    new_publisher_trusted_algorithms = remove_publisher_trusted_algorithm(
+        ddo.did, algorithm_ddo.did, publisher_ocean_instance.config.aquarius_url
+    )
+
+    assert new_publisher_trusted_algorithms is not None
+    assert len(new_publisher_trusted_algorithms) < len(publisher_trusted_algorithms)
+
+    # remove a trusted algorithm that does not belong to publisher_trusted_algorithms list
+    new_publisher_trusted_algorithms = remove_publisher_trusted_algorithm(
+        ddo.did, algorithm_ddo_v3.did, publisher_ocean_instance.config.aquarius_url
+    )
+
+    assert new_publisher_trusted_algorithms is not None
+    for _, trusted_algorithm in enumerate(publisher_trusted_algorithms):
```

Comment:
if you are not using the indices, use a simple for trusted_algorithm in publisher_trusted_algorithms. I think I did this modification already, but you force-pushed over my correction.